### ISSUE TITLE
Sphinx 2.0: napoleon extension fails to parse exception with tilde

### ISSUE THREAD

**Issue by RazerM** [sphinx-doc/sphinx#6220]
**Describe the bug**
Using the napoleon extension with `napoleon_numpy_docstring = False`, an exception with a tilde in front fails to parse.

I have a docstring

```python
"""
    Raises:
        ValueError: description.
        ~mymodule.MyError: description.
"""
```

This regex fails:

https://github.com/sphinx-doc/sphinx/blob/1499a97c28820ca2ff525d675e9c86fee917532d/sphinx/ext/napoleon/docstring.py#L103-L104

at this line:

https://github.com/sphinx-doc/sphinx/blob/1499a97c28820ca2ff525d675e9c86fee917532d/sphinx/ext/napoleon/docstring.py#L703

**To Reproduce**

```python
from sphinx.ext.napoleon.docstring import GoogleDocstring
assert GoogleDocstring._name_rgx.match('~mymodule.MyError') is None
```

**Expected behavior**
Sphinx renders it as if I used `:raises ~mymodule.MyError: description.`

**Environment info**
- Sphinx version: 2.0.0
- Sphinx extensions:  sphinx.ext.napoleon

**Additional context**
Bug introduced by #4046

**Comment by gmc444-b** [comment]
I found a similar error when parsing a section where the exception was enclosed in backquotes, ie:
``` Python
    Raises:
        `SomeError`
```

**Comment by RobRuana** [comment]
@tk0miya I must apologize. I no longer have the time to maintain Napoleon. I probably should have said something sooner.

**Comment by tk0miya** [comment]
@RobRuana Don't mind. I'll try to maintain it. Thank you for letting me know

**Comment by tk0miya** [comment]
I just posted #6237 to resolve this :-)

**Comment by tk0miya** [comment]
Fixed by #6237.
Thank you for reporting!
